### PR TITLE
First pass at variable quotas

### PR DESCRIPTION
**Pull request by hardillb**
part of #66 
## Description

Allows quota values to be passed from the Forge App.

It gets the quota sizes for the team when it checks the token passed.

## Related Issue(s)

#66 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Comment by hardillb** [comment]
This will depend on a second PR against the flowforge/flowforge project

**Review by knolleary — APPROVED** [review]

File: forge/auth.js [hunk 1]
```diff
@@ -21,22 +22,27 @@ module.exports = fp(async function (app, opts, done) {
...
         try {
             const token = getAuthToken(request)
             const cacheOk = checkCache(token, request.params.projectId)
             if (!cacheOk) {
-                if (!await checkToken(request.params.projectId, token)) {
+                const tokenResponse = await checkToken(request.params.projectId, token)
+                if (!tokenResponse) {
                     throw new Error('Invalid token')
                 }
+                request.quota = tokenResponse
```
Comment (knolleary) [inline forge/auth.js:37]:
```suggestion
                request.quota = tokenResponse.quota
```

File: forge/auth.js [hunk 2]
```diff
@@ -21,22 +22,27 @@ module.exports = fp(async function (app, opts, done) {
...
                 // update cache
                 authCache[token] = {
                     ttl: Date.now(),
-                    projectId: request.params.projectId
+                    projectId: request.params.projectId,
+                    quota: tokenResponse
```
Comment (knolleary) [inline forge/auth.js:42]:
```suggestion
                    quota: tokenResponse.quota
```

File: test/unit/setup.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ function authServer (config = {}) {
                 authToken = authConfig.find(auth => auth.projectId === projectId)
                 if (req.headers.authorization === ('Bearer ' + authToken.token)) {
                     res.writeHead(200)
-                    res.end('{}')
+                    res.end('{ "file": 1000, "context": 1000 }')
```
Comment (knolleary) [inline test/unit/setup.js:25]:
```suggestion
                    res.end('{ "quota": { "file": 1000, "context": 1000 } }')
```

**Review by knolleary — CHANGES_REQUESTED** [review]
To match the proposed changes in the server side, this wraps the expected response in a `quota` object

**Review by knolleary — APPROVED** [review]

**Pull request merged**